### PR TITLE
Print JS Exceptions to console.

### PR DESCRIPTION
**Pull request by PepsRyuu**
At the moment, if there's a mistake in the JavaScript code, there's no useful feedback about what went wrong making debugging difficult. This pull request provides the developer with a normal JavaScript exception error.

### CODE REVIEW

File: HoloJS/HoloJsHost/ScriptsLoader.h [hunk 1]
```diff
@@ -11,7 +16,7 @@ namespace HologramJS
 		concurrency::task<bool> LoadScripts(Windows::Storage::StorageFolder^ root, const std::wstring& jsonPath);
 		concurrency::task<bool> DownloadScripts(const std::wstring& uri);
 
-		const std::list<std::wstring>& LoadedScripts() { return m_loadedScripts; }
+		const std::list<Script*>& LoadedScripts() { return m_loadedScripts; }
```
Comment (Almost-Done) [inline HoloJS/HoloJsHost/ScriptsLoader.h]: This should be a unique or shared pointer to automatically free the allocated memory.

File: HoloJS/HoloJsHost/HologramJS.cpp [hunk 2]
```diff
@@ -155,16 +151,23 @@ HologramScriptHost::RunWebScriptApp(
...
-		}
+		ExecuteScripts(loader);
+
 	}
 
 	return true;
 }
 
+void 
+HologramScriptHost::ExecuteScripts(std::unique_ptr<HologramJS::ScriptsLoader>& loader)
```
Comment (Almost-Done) [inline HoloJS/HoloJsHost/HologramJS.cpp]: Maybe this should be a method of the ScriptsLoader; that way there is no need to expose the loaded scripts, just the Execute method to execute them all.

File: HoloJS/HoloJsHost/ScriptsLoader.cpp [hunk 3]
```diff
@@ -97,7 +100,10 @@ ScriptsLoader::LoadScripts(
 			wstring scriptText = scriptPlatformString->Data();
 			if (!scriptText.empty())
 			{
-				m_loadedScripts.emplace_back(std::move(scriptText));
+				Script* script = new Script();
```
Comment (Almost-Done) [inline HoloJS/HoloJsHost/ScriptsLoader.cpp]: This won't merge properly using auto merge - it does not compile. I added a case above this code for downloading scripts if an absolute URL is given; the new code needs to be given the same treatment. Could you please merge master

**Comment by PepsRyuu** [comment]
Updated! :)

**Comment by Almost-Done** [comment]
Thank you for this code contribution. It's really nice to see JavaScript exceptions when debugging native code.

As a side-effect of this change, when Script debugging you can now see script names in Solution Explorer. Very cool!

image

image

**Pull request merged**